src/Model/Contact.php

```diff
@@ -136,6 +136,31 @@ public static function selectFirst(array $fields = [], array $condition = [], ar
 		return $contact;
 	}
 
+	/**
+	 * Insert a row into the contact table
+	 * Important: You can't use DBA::lastInsertId() after this call since it will be set to 0.
+	 *
+	 * @param array        $param               parameter array
+	 * @param bool         $on_duplicate_update Do an update on a duplicate entry
+	 *
+	 * @return boolean was the insert successful?
+	 * @throws \Exception
+	 */
+	public static function insert($param, $on_duplicate_update = false)
+	{
+		$ret = DBA::insert('contact', $param, $on_duplicate_update);
+		$contact = DBA::selectFirst('contact', ['nurl', 'uid'], ['id' => DBA::lastInsertId()]);
+		if (!DBA::isResult($contact)) {
+			// Shouldn't happen
```

Comment:
Maybe a good logpoint to get insights in case it does happen?

Reply:
I don't see a reason why. Even when this fails (e.g. at duplicated entries) it doesn't harm anything.

Reply:
Ok